documentation/en/chain-indexer-overview-for-operators.md

```diff
@@ -0,0 +1,415 @@
+
+```toml
+[Fevm]
+# Enable the ETH RPC APIs.
+# This is not required for ChainIndexer support, but ChainIndexer is required if you enable this.
+  EnableEthRPC = true
+
+[Events]
+# Enable the Actor Events APIs.
+# This is not required for ChainIndexer support, but ChainIndexer is required if you enable this.
+  EnableActorEventsAPI = true
+
+[ChainIndexer]
+# Enable the ChainIndexer, which is required for the ETH RPC APIs and Actor Events APIs.
+# If they are enabled, but the ChainIndexer is not, Lotus will exit during startup.
+# (ChainIndexer needs to be explicitly enabled to signal to the node operator the extra
+# supporting functionality that will now be running.)
+  EnableIndexer = true 
+```
+
+You can learn more about these configuration options and other configuration options available for the `ChainIndexer` [here](https://github.com/filecoin-project/lotus/blob/master/documentation/en/default-lotus-config.toml).
+
+
+### Garbage Collection
+
+The `ChainIndexer` includes a garbage collection (GC) mechanism to manage the amount of historical data retained.  See the [ChainIndexer size requirements](#backfill-disk-space-requirements).
+
+By default, GC is disabled to preserve all indexed data.
+
+To configure GC, use the `GCRetentionEpochs` parameter in the `ChainIndexer` section of your config.
+
+The ChainIndexer [periodically runs](https://github.com/filecoin-project/lotus/blob/master/chain/index/gc.go#L15) GC if `GCRetentionEpochs` is > 0 and removes indexed data for epochs older than `(current_head_height - GCRetentionEpochs)`.
+
+```toml
+[ChainIndexer]
+  GCRetentionEpochs = X  # Replace X with your desired value
+```
+
+- Setting `GCRetentionEpochs` to 0 (**default**) disables GC.
+- Any positive value enables GC and determines the number of epochs of historical data to retain.
+
+#### Recommendations
+
+1. **Archival Nodes**: **Keep GC disabled** (`GCRetentionEpochs` = 0) to retain all indexed data.
+
+2. **Non-Archival Nodes**:  Set `GCRetentionEpochs` to match the amount of chain state your node retains 
+
+**Example:** if your node is configured to retain Filecoin chain state with a Splitstore Hotstore that approximates 2 days of epochs, set `GCRetentionEpochs` to at least `retentionDays * epochsPerDay = 2 * 2880 = 5760`).
+
+**Warning:** Setting this value below the chain state retention period may degrade RPC performance and reliability because the ChainIndexer will lack data for epochs still present in the chain state.
+
+**Note:** `Chainstore.Splitstore` is configured in terms of bytes (not epochs) and `ChainIndexer.GCRetentionEpochs` is in terms of epochs (not bytes).  For the purposes of this discussion, we're assuming operators have determined `Chainstore.Splitstore.HotStoreMaxSpaceTarget` and `Chainstore.Splitstore.HotStoreMaxSpaceThreshold` values that approximate a certain number days of storage in the Splitstore Hotstore.  The guidance here is to make sure this approximation exceeds `ChainIndexer.GCRetentionEpochs`.
```

Comment:
@rvagg @BigLep 

Have we checked the logic related to `HotStoreMaxSpaceTarget` and `HotStoreMaxSpaceThreshold` to be absolutely certain of what we're talking about here ? 

Unless we're **_absolutely_** certain that this is the only splitstore config we're trying to approximate in `GCRetentionEpochs` -> I honestly think this section will confuse RPC providers using (snapshots + splitstore) more than helping them.

Will defer the final decision to you'll as I do not know enough about the splitstore but I do know that RPC providers also do not know much about the splitstore. Please do give it a thought.

Reply:
All I know is that `GCRetentionEpochs` doesn't have a direct relationship to the way splitstore works so we shouldn't mislead in specifying a timeframe that matches splitstore.

Perhaps @ZenGround0 could help us here—would you mind sanity checking this line, do you think it's a reasonable statement given your knowledge of splitstore?

Reply:
@rvagg 

So why not just say "make sure the GCRetentionEpochs epoch should be in line with how much chain state is retained on your node/how long back in the past RPC queries are served for".

RPC providers have good operational expertise and know things such as "My node only serves requests for the past 16 hours of epochs" or "My node only retains 3 days of chain state" and so can align the `GCRetentionEpochs` param with that.

I really don't think we should be taking them down a journey of `HotStoreMaxSpaceTarget ` etc etc here -> it will confuse them as I'm fairly confident they wont know what to do with this information.

Reply:
I don't have strong feelings here so will default to you guys.  That said, with a free re-read, I am still good with what's written.  This is because it helps draw attention that:
1. there are multiple configuration knobs that a splitstore/hostore-using operator needs to pay attention to AND 
2. that splitstore/hotstore configuration is in terms of bytes but ChainIndexer configuration is in terms of epochs

I believe those are the main points to drive home.

I agree that splitstore config and operations feel complex as I read https://github.com/filecoin-project/lotus/blob/master/documentation/en/default-lotus-config.toml#L151 , but after I got my head into the mental model of how it works based on the comments there, I think calling out `Chainstore.Splitstore.HotStoreMaxSpaceTarget` and `Chainstore.Splitstore.HotStoreMaxSpaceThreshold` seems reasonable.  (If the code doesn't match what's described in the comments, then I don't know what to say...)

Anyways, I will leave this as is unless you guys insist otherwise or we get operator feedback on this.  (Feel free to make changes too.)